app/src/main/java/lt/vilnius/tvarkau/utils/GoogleSignInHelper.java

```diff
@@ -0,0 +1,68 @@
+package lt.vilnius.tvarkau.utils;
+
+import android.content.Intent;
+import android.support.v4.app.FragmentActivity;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.Auth;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.auth.api.signin.GoogleSignInResult;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+
+import lt.vilnius.tvarkau.MainActivity;
+import lt.vilnius.tvarkau.R;
+import lt.vilnius.tvarkau.entity.Profile;
+
+
+public class GoogleSignInHelper implements GoogleApiClient.OnConnectionFailedListener {
+    public static final int RC_SIGN_IN = 9001;
+
+    private GoogleApiClient googleApiClient;
+
+    private FragmentActivity activity;
+
+
+    public GoogleSignInHelper(FragmentActivity activity) {
```

Comment:
I don't like passing the whole `FragmentActivity` here. I think it would make sense to create separate `interface` with logged in callbacks  (or use callbacks from `GoogleApiClient`) and pass it to GoogleSignHelper. It's really error prone here, because `onActivityResult` of `FragmentActivity` can be easily messed up.